### PR TITLE
#282: get error extension `code` from exception name or annotation

### PR DESCRIPTION
**Pull request by t1**

### CODE REVIEW

**Comment by t1** [comment]
@andymc12 sorry to add changes to an already approved PR, but I thought that TCK tests would be a good idea to make it easier to later migrate that to MP.

File: client/generator-test/pom.xml [hunk 1]
```diff
@@ -45,9 +45,8 @@
             <version>${project.version}</version>
         </dependency>
         <dependency>
-            <groupId>io.smallrye</groupId>
-            <artifactId>smallrye-graphql-api</artifactId>
-            <version>${project.version}</version>
+            <groupId>org.eclipse.microprofile.graphql</groupId>
+            <artifactId>microprofile-graphql-api</artifactId>
```
Comment (phillip-kruger) [inline client/generator-test/pom.xml:49]: Why did this change ?

File: server/api/src/main/java/io/smallrye/graphql/api/GraphQlErrorCode.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Retention(RUNTIME)
+@Target({ TYPE, ANNOTATION_TYPE })
+public @interface GraphQlErrorCode {
```
Comment (phillip-kruger) [inline server/api/src/main/java/io/smallrye/graphql/api/GraphQlErrorCode.java:12]: Remove GraphQl (to be consistent with other annotations) So just `@ErrorCode`

File: server/implementation/src/main/java/io/smallrye/graphql/execution/error/ExecutionErrorsService.java [hunk 3]
```diff
@@ -86,12 +86,25 @@ private JsonObject toJsonError(GraphQLError error) {
...
         addKeyValue(objectBuilder, EXCEPTION, exception.getClass().getName());
         addKeyValue(objectBuilder, CLASSIFICATION, error.getErrorType().toString());
+        addKeyValue(objectBuilder, CODE, toErrorCode(exception));
         Map<String, Object> extensions = error.getExtensions();
         populateCustomExtensions(objectBuilder, extensions);
 
         return Optional.of(objectBuilder.build());
     }
 
+    private String toErrorCode(Throwable exception) {
```
Comment (phillip-kruger) [inline server/implementation/src/main/java/io/smallrye/graphql/execution/error/ExecutionErrorsService.java:96]: Please add a TODO that this should change to use the Model and the model needs to get this using Jandex.

Comment (t1) [reply server/implementation/src/main/java/io/smallrye/graphql/execution/error/ExecutionErrorsService.java:96]: done

File: server/api/src/main/java/io/smallrye/graphql/api/GraphQlErrorCode.java (hunk 2 above)
Comment (t1) [reply server/api/src/main/java/io/smallrye/graphql/api/GraphQlErrorCode.java]: done

File: client/generator-test/pom.xml (hunk 1 above)
Comment (t1) [reply client/generator-test/pom.xml]: reverted

**Pull request merged**